packages/ckeditor5-dev-changelog/src/utils/getreleasetypefromversion.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @license Copyright (c) 2003-2025, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import { type ChangelogReleaseType } from '../types.js';
+import semver from 'semver';
+
+export function getReleaseTypeFromVersion( currentVersion: string, nextVersion: string ): ChangelogReleaseType {
+	const currentVersionPrerelease = semver.prerelease( currentVersion );
+	const nextVersionPrerelease = semver.prerelease( nextVersion );
+
+	if ( !nextVersionPrerelease ) {
+		return 'latest';
+	}
+
+	if ( nextVersionPrerelease[ 0 ] === currentVersionPrerelease?.[ 0 ] ) {
+		return 'prerelease';
+	}
+
+	return 'prerelease-promote';
```

Comment:
To avoid optional chaining operators, which, in my opinion, can be challenging to understand, I would apply default values and reorganize the logic slightly.

```suggestion
	const [ currentChannel ] = semver.prerelease( currentVersion ) || [ 'latest' ];
	const [ nextChannel ] = semver.prerelease( nextVersion ) || [ 'latest' ];

	if ( nextChannel === 'latest' ) {
		return 'latest';
	}

	if ( nextChannel === currentChannel ) {
		return 'prerelease';
	}

	return 'prerelease-promote';
```

Reply:
Added in [dd6e559](https://github.com/ckeditor/ckeditor5-dev/pull/1183/commits/dd6e559bd34fc6ff9a3a0e08222ca410cc60764f)